### PR TITLE
Rename Json protocol to NewtonsoftJson

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/aspnet/SignalR/issues/3308

We talked about removing NewtonsoftJson protocol from being on by default, do we want to make that change?

### CODE REVIEW

File: src/SignalR/run.ps1 [hunk 1]
```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env powershell
```
Comment (davidfowl) [inline src/SignalR/run.ps1]: Whats this?

File: src/SignalR/run.cmd [hunk 2]
```diff
@@ -0,0 +1,2 @@
+@ECHO OFF
```
Comment (natemcmaster) [inline src/SignalR/run.cmd]: Don't copy the run.{cmd,sh,ps1} scripts here. Instead, please add scripts that look like these:

https://github.com/aspnet/AspNetCore/blob/master/src/Razor/build.cmd
https://github.com/aspnet/AspNetCore/blob/master/src/Razor/build.sh

Comment (BrennanConroy) [reply src/SignalR/run.cmd]: Oh, is that already working?

Comment (natemcmaster) [reply src/SignalR/run.cmd]: It should work.

File: src/SignalR/test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs [hunk 3]
```diff
@@ -119,7 +119,7 @@ public void WriteMessage(string protocolTestDataName)
 
             var expectedOutput = Frame(testData.Json);
 
-            var protocolOptions = new JsonHubProtocolOptions
+            var protocolOptions = new NewtonsoftJsonHubProtocolOptions
```
Comment (JamesNK) [inline src/SignalR/test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs]: Rename JsonHubProtocolTests.cs -> NewtonsoftJsonHubProtocolTests.cs

**Review by JamesNK — APPROVED** [review]

File: src/SignalR/test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs (hunk 3 above)
Comment (BrennanConroy) [reply src/SignalR/test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs]: I was going to use this file for both NewtonsoftJson and System.Text.Json when we add that.

**Review by halter73 — APPROVED** [review]

File: src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.Json/baseline.netcore.json [hunk 4]
```diff
@@ -1,253 +0,0 @@
-{
-  "AssemblyIdentity": "Microsoft.AspNetCore.SignalR.Protocols.Json, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60",
```
Comment (halter73) [inline src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.Json/baseline.netcore.json]: Are these still be verified? Should we generate a new baseline?

File: src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson.csproj [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Description>Implements the SignalR Hub Protocol over JSON.</Description>
+    <Description>Implements the SignalR Hub Protocol using JSON over Newtonsoft.Json.</Description>
```
Comment (halter73) [inline src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson.csproj]: nit: "over Newtonsoft.Json" sounds weird. Maybe simplify to just "using Newtonsoft.Json"

File: src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson/Protocol/NewtonsoftJsonHubProtocol.cs [hunk 6]
```diff
@@ -17,9 +17,9 @@
 namespace Microsoft.AspNetCore.SignalR.Protocol
 {
     /// <summary>
-    /// Implements the SignalR Hub Protocol using JSON.
+    /// Implements the SignalR Hub Protocol using JSON over Newtonsoft.Json.
```
Comment (halter73) [inline src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson/Protocol/NewtonsoftJsonHubProtocol.cs]: ditto about "over Newtonsoft.Json"

File: src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.Json/baseline.netcore.json (hunk 4 above)
Comment (BrennanConroy) [reply src/SignalR/src/Microsoft.AspNetCore.SignalR.Protocols.Json/baseline.netcore.json]: I can probably copy the same baseline file and change the types manually. The methods and everything should stay the same.

**Pull request merged**